### PR TITLE
Initial working version of uploading assignments

### LINKED ISSUE #1: Pushing Assignments
**Is your feature request related to a problem? Please describe.**
It can be quite frustrating to have to use the website to repeatedly submit files to Gradescope, especially if you want to submit many times or if you want to check with the autograder. I would appreciate if there was a way to use the Gradescope API to submit any number of files to a particular assignment in a class; 

**Describe the solution you'd like**
You could specify the Course ID and Assignment ID and then pass any number of files, possible with a --files flag. This may be out of scope but it could also be convenient to have the Gradescope CLI wait until it gets a response back and show the passed/failed testcases.

### PR DESCRIPTION
**Pull request by calvinatian**
### Summary

Adds feature to upload assignments

### Checks

- [x] Tested changes
- [ ] Attached Logs

### Reference to the issue

Resolves #1

### CODE REVIEW

File: src/sylveon/_classes/_upload.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    leaderboard_name: str | None = None,
+) -> str | None:
+    """Uploads given file objects to the specified assignment on Gradescope.
+
+    Args:
+        session (requests.Session): The session object to use for making HTTP requests.
+        course_id (str): The ID of the course on Gradescope.
+        assignment_id (str): The ID of the assignment on Gradescope.
+        *files (io.TextIOWrapper): Variable number of file objects to upload.
+        leaderboard_name (str | None, optional): The name of the leaderboard. Defaults to None.
```
Comment (KevZhe) [inline src/sylveon/_classes/_upload.py:25]: just curious, what exactly is the leaderboard?

File: src/sylveon/_classes/_upload.py
Comment (KevZhe) [inline src/sylveon/_classes/_upload.py:1]: looks good

**Review by KevZhe — APPROVED** [review]

File: src/sylveon/_classes/_upload.py (hunk 1 above)
Comment (calvinatian) [reply src/sylveon/_classes/_upload.py:25]: Leaderboards can be set up for programming assignments based on the autograder for programming assignments: https://gradescope-autograders.readthedocs.io/en/latest/leaderboards/

To be honest, I've never been in a course that used the feature. But for the test assignment I enabled it to play around with it.

**Comment by jmargaret2** [comment]
fixed the merge conflict, you had uploaded the requirements.txt with some half-baked changes, fixed those and will merge

**Pull request merged**